src/test/javascript/spec/component/programming-exercise/programming-exercise-instructor-submission-state.component.spec.ts

```diff
@@ -51,11 +51,21 @@ describe('ProgrammingExerciseInstructorSubmissionStateComponent', () => {
         return debugElement.query(By.css(resultEtaId));
     };
 
-    beforeEach(async () => {
+    beforeEach(() => {
         return TestBed.configureTestingModule({
-            imports: [TranslateModule.forRoot(), ArtemisTestModule, ArtemisProgrammingExerciseActionsModule],
+            imports: [ArtemisTestModule],
+            declarations: [
+                ProgrammingExerciseInstructorSubmissionStateComponent,
+                ProgrammingExerciseTriggerAllButtonComponent,
+                ButtonComponent,
+                FeatureToggleDirective,
+                FeatureToggleLinkDirective,
```

Comment:
These four components are not mocked. Is this required for the tests to work? Otherwise mocking them might further increase the performance

Reply:
Yes, they are required. The tests include some checks related to those components as well and if we want to mock them I should delete some of the tests. And I think this will decrease the coverage, that's why I didn't do it.

Those components were not in the declarations before, because another test module was used in the imports which included much more not mocked components, that were not really needed for the current test.

Reply:
[nit] But shouldn't we have a separate test file for every component instead?
Especially for components like `ButtonComponent` that seem to be tightly related to the `ProgrammingExercise`

Reply:
I agree with you, but since the test was already created like that I didn't want to do that change